src/components/app/gs-scheduler/RequestForm/ContactRequestForm.tsx

```diff
@@ -0,0 +1,179 @@
+import React, {useEffect} from "react";
+import {Controller, FormProvider, useForm} from "react-hook-form";
+import { Button } from "@/components/ui/button";
+import { zodResolver } from "@hookform/resolvers/zod";
+import "react-datepicker/dist/react-datepicker.css";
+import { Separator } from "@/components/ui/separator";
+import { TimePickerField } from "@/components/ui/wrapper/timepickerfield";
+import { locations } from "@/api/gs-locations";
+import FormFieldWrapper from "@/components/ui/wrapper/formfieldwrapper";
+import CheckboxFieldWrapper from "@/components/ui/wrapper/checkboxfieldwrapper";
+import { ContactRequestFormSchema, ContactRequestFormData } from "./ContactRequestFormSchema";
+import Combobox from "@/components/ui/combobox";
+import { satellites } from "@/api/gs-satellites";
+import {formatToISOString} from "@/lib/formatToISOString";
+import axios from "axios";
+import {toast} from "@/hooks/use-toast";
+
+interface ContactRequestFormProps {
+    location: typeof locations[0];
+}
+
+const satelliteOptions = satellites
+    .filter(sat => sat.satellite_id && sat.label)
+    .map((sat) => ({
+        value: sat.satellite_id.toString(),
+        label: sat.label,
+    }));
+
+const ContactRequestForm: React.FC<ContactRequestFormProps> = ({ location }) => {
+    useEffect(() => {
+        console.log("ContactRequestForm: Location updated to", location.label);
+    }, [location]);
+
+    const form = useForm<ContactRequestFormData>({
+        resolver: zodResolver(ContactRequestFormSchema),
+        defaultValues: {
+            missionName: "",
+            satelliteId: satelliteOptions[0]?.value || "",
+            orbit: "",
+            uplink: false,
+            telemetry: false,
+            science: false,
+            aosTime: undefined,
+            rfOnTime: undefined,
+            rfOffTime: undefined,
+            losTime: undefined,
+        },
+    });
+
+    const onSubmit = async (values: ContactRequestFormData) => {
+        const payload = {
+            ...values,
+            aosTime: formatToISOString(values.aosTime),
+            rfOnTime: formatToISOString(values.rfOnTime),
+            rfOffTime: formatToISOString(values.rfOffTime),
+            losTime: formatToISOString(values.losTime),
+            location: location.label,
+            satellite_id: values.satelliteId,
+        };
+        // Send the request to the backend using axios
+        try {
+            const response = await axios.post(process.env.NEXT_PUBLIC_API_BASE_URL + '/api/v1/request/contact', payload);
```

Comment:
check for status code 200 before accepting it (not sure if try/catch catches codes like that)

Reply:
ahh didnt know that